crates/weaver_forge/src/file_loader.rs

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Set of supported template loaders
+
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+use minijinja::ErrorKind;
+use walkdir::WalkDir;
+
+use crate::error::Error;
+use crate::error::Error::TargetNotSupported;
+
+/// An abstraction for loading files from a file system, embedded directory, etc.
+pub trait FileLoader: Send + Sync {
```

Comment:
Nit: Can you add the `Send + Sync` requirement only in the `weaver_forge` where you need the type to be send+sync?

Reply:
See [9b8e7a7](https://github.com/open-telemetry/weaver/pull/171/commits/9b8e7a7564c059510f63b361f8336b485f5ef12c)